### PR TITLE
Fix for panic when removing RenetServer or RenetClient resource

### LINKED ISSUE #7: Panic if you try to remove RenetClient or RenetServer resources after registering events
## Description
If you try to remove RenetClient or RenetServer resources after registering client or/and? server events.

Example of the panic when removing RenetServer:
```
thread 'Compute Task Pool (9)' panicked at 'Resource requested by bevy_replicon::network_event::client_event::receiving_system<leafwing_input_manager::action_state::ActionDiff<stellar_squeezebox:
:player::PlayerAction, stellar_squeezebox::network::NetworkOwner>> does not exist: renet::server::RenetServer', C:\Users\Paul\.cargo\registry\src\github.com-1ecc6299db9ec823\bevy_ecs-0.10.1\src\system\system_param.rs:555:17
```

## What I Expected
I expected this to not panic and instead drop any resources needed to close the connection. This would allow setting up a new connection, for example when the user want's to stop playing one game they joined and join another or host their own.

## Steps to replicate
```shell
git clone https://github.com/paul-hansen/bevy-jam-3.git
git checkout cb451e27
cargo run -- --listen 127.0.0.1
cargo run -- --connect 127.0.0.1
```
On either the client or the server press escape and "y" to accept. It will panic here trying to run the event systems saying the client or server resource does not exist.

I have a working patched version of bevy_replicon. If you want to test with that version, do the same but checkout `4217a6d5` instead. (Our game's main branch is using this patch too) Instead of a panic you should see a dialog that allows you to join or create a new game.

### PR DESCRIPTION
**Pull request by paul-hansen**
Fixes #7

### CODE REVIEW

**Review by Shatur — CHANGES_REQUESTED** [review]
Thanks! Could you try to put these systems to `CoreSet::StateTransition` instead and apply `.before(run_enter_schedule::<S>)`? This will make it obvious and comments won't be necessary.
I also think that you could call `.in_base_set` on a tuple to avoid calling it on each system.

**Comment by paul-hansen** [comment]
Yeah that's much clearer. I hadn't noticed CoreSet::StateTransitions before, learned some things, thanks!

File: src/client.rs [hunk 1]
```diff
@@ -22,11 +23,17 @@ impl Plugin for ClientPlugin {
...
+                )
+                    .before(run_enter_schedule::<ClientState>)
+                    .in_base_set(CoreSet::StateTransitions),
+            )
             .add_systems((
-                Self::no_connection_state_system.run_if(resource_removed::<RenetClient>()),
-                Self::connecting_state_system
-                    .run_if(bevy_renet::client_connecting)
-                    .in_set(OnUpdate(ClientState::NoConnection)),
                 Self::connected_state_system
```
Comment (Shatur) [inline src/client.rs:37]: Could you please put this system to `CoreSet::StateTransitions` too?
```suggestion
                Self::connected_state_system
```

**Review by Shatur — CHANGES_REQUESTED** [review]
One minor suggestion?

**Review by Shatur — APPROVED** [review]
Thank you a lot!

**Comment by Shatur** [comment]
Test fails, I will take a look.

**Comment by paul-hansen** [comment]
Hmm also looking, just running `cargo test` works locally for me.

**Comment by Shatur** [comment]
Looks like it happens sometimes.

**Comment by paul-hansen** [comment]
Ah yeah I did get it after a few more tries.

**Comment by Shatur** [comment]
Fixed it. In tests we had two `app.update()` because of the one frame delay. Not longer needed!

**Comment by paul-hansen** [comment]
Nice!

**Pull request merged**